src/Paket.PowerShell/ArgumentTabCompletion.ps1

```diff
@@ -0,0 +1,36 @@
+﻿
+# http://stackoverflow.com/questions/30923696/add-custom-argument-completer-for-cmdlet
+
+# https://github.com/mariuszwojcik/RabbitMQTools/blob/master/TabExpansions.ps1
+function createCompletionResult([string]$text, [string]$value, [string]$tooltip) {
+    if ([string]::IsNullOrEmpty($value)) { return }
+    if ([string]::IsNullOrEmpty($text)) { $text = $value }
+    if ([string]::IsNullOrEmpty($tooltip)) { $tooltip = $value }
+    $completionText = @{$true="'$value'"; $false=$value  }[$value -match "\W"]
+    $completionText = $completionText -replace '\[', '``[' -replace '\]', '``]'
+    New-Object System.Management.Automation.CompletionResult $completionText, $text, 'ParameterValue', $tooltip | write
+}
+
+$findPackages = {
+    param($commandName, $parameterName, $wordToComplete, $commandAst, $fakeBoundParameter)
+    Paket-FindPackages -SearchText $wordToComplete -Max 100 | % {
+        createCompletionResult $_ $_ $_ | write
+    }
+}
+
+$findPackageVersions = {
+    param($commandName, $parameterName, $wordToComplete, $commandAst, $fakeBoundParameter)
+    if (-not $fakeBoundParameter.NuGet){ return }
+    Paket-FindPackageVersions -Name $fakeBoundParameter.NuGet -Max 100 | % {
+        createCompletionResult $_ $_ $_ | write
+    }
+}
+
+# create and add $global:options to the list of completers
+# http://www.powertheshell.com/dynamicargumentcompletion/
+if (-not $global:options) { $global:options = @{CustomArgumentCompleters = @{};NativeArgumentCompleters = @{}}}
+
+$global:options['CustomArgumentCompleters']['Paket-Add:NuGet'] = $findPackages
+$global:options['CustomArgumentCompleters']['Paket-Add:Version'] = $findPackageVersions
```

Comment:
More argument tab completers can be added following this pattern. It must be done in PowerShell. The output of another command is used to do the tab completion.